doc/shape-validation.md

```diff
@@ -32,15 +32,15 @@ Each shape item follows these rules:
 
 * an ``Integer`` indicates a fix dimension for the same item
 
-* a ``~``, `:` or `None` indicates a single dimension of arbitrary length.
```

Comment:
`:` only works inside the python functions but it is not valid yaml syntax

I removed the mention of `:` here since those function are not supposed to be called directly and it is confusing